Framework/Algorithms/inc/MantidAlgorithms/Integration.h

```diff
@@ -69,6 +69,9 @@ class MANTID_ALGORITHMS_DLL Integration : public API::Algorithm {
   /// Create the outputworkspace
   API::MatrixWorkspace_sptr getOutputWorkspace(API::MatrixWorkspace_sptr inWS, const int minSpec, const int maxSpec);
   std::map<std::string, std::string> validateInputs() override;
+  void integrateSpectrum(const API::ISpectrum &inSpec, API::ISpectrum &outSpec, const std::vector<double> *Fin,
```

Comment:
LGTM, my only nits with this are:
- Can allocate a vector on the stack and pass in by ref and just check for .empty as the sentinel
- Clang-tidy will complain about `const` scalar types which are passed by copy, can we drop const from double and bool?

Reply:
No problem about removing the `const`s. Re your first point, I could do this for Fin but I think I have to leave Fout as a pointer because I need to update the data inside rebinned_output->dataF(). So seemed more symmetrical to leave Fin and Fout as pointers? Knew you'd like a bit of raw pointer action...

Reply:
That sounds good to me, we'll stick with the raw pointer squad so we can get this out of the way for the weekend :laughing: